### DISCUSSION TITLE
Multi window support with a master store

### DISCUSSION THREAD

**Discussion by ccy-213** [tauri-apps/tauri#11782] [General]
I have a web app built using vue js with vuex as state management.my app connecting to server via websocket. I would like to make it a desktop application, and support multiple windows. Is there any guidelines on how can I keep a single view store in main window that can be sync across all sub windows?

**Comment by FabianLars** [comment]
We don't enough control over the webview to have actually shared state, so the solution to your question is the same as how to share state across browser windows/tabs (that should hopefully make asking google etc easier).
That said, in Tauri you could use the rust side to share state (via tauri::State in commands or events for example) but there's no integration for vuex so that would have to be wired up manually.

**Comment by konsalex** [comment]
@ccy-213 we recently wrote a piece around this topic with React specifically, but you could maybe use this approach if you do not care about loose sync:
https://gethopp.app/blog/tauri-window-state-sync
